src/components/user/createAppointment/CreateAppointment.js

```diff
@@ -26,14 +27,14 @@ const CreateAppointment = () => {
   const [disabled, setDisabled] = useState(true);
 
   useEffect(() => {
-    if (calendarData && time && formData) {
+    if (calendarData && time && formData.doctorId && formData.reason) {
       setDisabled(false);
     }
   });
```

Comment:
Read about second param in useEffect - []

Reply:
set as dependencies [calendarData, time, formData]. I didn't notice it(